src/bun.js/test/jest.zig

```diff
@@ -61,6 +63,161 @@ fn notImplementedProp(
     return null;
 }
 
+pub const DiffFormatter = struct {
+    received: JSValue,
+    expected: JSValue,
+    globalObject: *JSC.JSGlobalObject,
+    not: bool = false,
+
+    pub fn format(this: DiffFormatter, comptime _: []const u8, _: std.fmt.FormatOptions, writer: anytype) !void {
+        var received_buf = MutableString.init(default_allocator, 0) catch unreachable;
+        var expected_buf = MutableString.init(default_allocator, 0) catch unreachable;
+        defer {
+            received_buf.deinit();
+            expected_buf.deinit();
+        }
+
+        {
+            var buffered_writer_ = bun.MutableString.BufferedWriter{ .context = &received_buf };
+            var buffered_writer = &buffered_writer_;
+
+            var buf_writer = buffered_writer.writer();
+            const Writer = @TypeOf(buf_writer);
+
+            JSC.ZigConsoleClient.format(
```

Comment:
we probably should make the options to `format` a Struct because its getting a lot of fields

Reply:
👍